components/tooltip/script.js

```diff
@@ -0,0 +1,103 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Coherent Labs AD. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import components from 'coherent-gameface-components';
+import template from './template.html';
+const TOOLTIP_MARGIN = 10;
+
+class Tooltip extends HTMLElement {
+    constructor() {
+        super();
+        this.template = template;
+
+        this.visible = false;
+    }
+
+    connectedCallback() {
+        this.position = this.getAttribute('position') || 'top';
+        this.showOn = this.getAttribute('on') || 'click';
+        this.hideOn = this.getAttribute('off') || 'click';
+        this.elementSelector = this.getAttribute('target');
+
+        this.triggerElement = document.querySelector(this.elementSelector);
+        if (!this.triggerElement) {
+            console.error(`An element with selector ${this.elementSelector} does not exit. Please make sure the selector is correct and the element exists.`);
+            return;
+        }
+
+        components.loadResource(this)
+            .then((result) => {
+                this.template = result.template;
+                components.renderOnce(this);
+                this.attachEventListeners();
+            })
+            .catch(err => console.error(err));
+    }
+
+    attachEventListeners() {
+        if (this.showOn === this.hideOn) {
+            this.triggerElement.addEventListener(this.showOn, (e) => this.toggle(e));

```

Comment:
```suggestion
            this.triggerElement.addEventListener(this.showOn, () => this.toggle());
```
?
Same for the `addEventListener`s below.

Reply:
Yes. Removed.